rclcpp/src/rclcpp/time.cpp

```diff
@@ -109,7 +109,7 @@ Time::operator=(const Time & rhs) = default;
 Time &
 Time::operator=(const builtin_interfaces::msg::Time & time_msg)
 {
-  *this = Time(time_msg);
+  *this = Time(time_msg, get_clock_type());
```

Comment:
> The assignment operator from builtin_interfaces::msg::Time no longer overrides the current clock type. I do not know why this is the default behavior

is this because of this? `RCL_SYSTEM_TIME` will be returned from `this->get_clock_type()`? that i think better behavior for user application, to change implicitly clock type into `RCL_SYSTEM_TIME`.

Reply:
by the way, `time.hpp` line 96-97 doc section needs to be updated.